release-tools/stochss_ami_manager.py

```diff
@@ -0,0 +1,547 @@
+            tmp_log_stdout_file.close()
+
+            if output == "Instance {0} with ip {1} is up!".format(self.instance_id, self.instance_ip):
+                print output
+                break
+
+    def __reboot_instance(self):
+        print '=================================================='
+        print 'Rebooting Instance {0}...'.format(self.instance_id)
+        self.ec2_connection.reboot_instances(instance_ids=[self.instance_id])
+        self.__wait_until_successful_ssh()
+
+
+    def __install_python_packages(self):
+        header = 'Installing Python Packages...'
+        print '=================================================='
+        print header
+
+        commands = []
+        for package in self.python_packages:
+            if package.has_key('version'):
+                commands.append('sudo pip uninstall -y {0}'.format(package['name']))
+                commands.append('sudo pip install {0}=={1}'.format(package['name'], package['version']))
+            else:
+                commands.append('sudo pip install {0}'.format(package['name']))
+
+        command = ';'.join(commands)
+        self.__run_remote_command(command=command, log_header=header)
+
+
+    def __download_stochss_repo(self):
+        header = 'Downloading StochSS...'
+        print '=================================================='
+        print header
+        commands = ['rm -rf stochss',
+                    'git clone --recursive {0}'.format(self.git_repo['url'])]
+
+        if self.git_repo.has_key('branch'):
+            commands.append('cd stochss')
+            commands.append('git checkout {0}'.format(self.git_repo['branch']))
+
+        command = ';'.join(commands)
+        self.__run_remote_command(command=command, log_header=header)
+
+    def __compile_stochss(self):
+        header = 'Compiling StochSS...'
+        print '=================================================='
+        print header
+        commands = ['cd stochss',
+                    './run.ubuntu.sh --install']
+        command = ';'.join(commands)
+        self.__run_remote_command(command=command, log_header=header)
+
+    def __extra_steps_for_old_amis(self):
+        header = 'Extra Steps for Old AMIs'
+        print '=================================================='
+        print header
+        commands = ['ln -s stochss/ode ode',
+                    'ln -s stochss/StochKit StochKit',
+                    'ln -s stochss/stochoptim stochoptim',
+                    'ln -s stochss/pyurdme/pyurdme_wrapper.py pyurdme_wrapper.py',
+                    'ln -s stochss/app/lib/pyurdme-stochss pyurdme',
+                    'ln -s stochss/app/backend/sccpy.py sccpy.py',
+                    'ln -s stochss/app/backend/tasks.py tasks.py',
+                    "sed -i.bak $'s/import pyurdme/import sys\\\\\\nsys.path.append(\\'\/home\/ubuntu\/pyurdme\\')\\\\\\nsys.path.append(\\'\/home\/ubuntu\/stochss\/app\\')\\\\\\nimport pyurdme/g' pyurdme_wrapper.py"]
```

Comment:
This line is dangerous, it assumes we won't change pyurdme_wrapper.py, can we do this another way?

Also, I am confused about the necessity of this function.  Are these extra steps always needed?  It seems to be called every time based on lines 106,155-6.